force-app/main/default/classes/MergeFieldsClass.cls

```diff
@@ -136,18 +168,156 @@ public with sharing class MergeFieldsClass {
         return strList;    
     }
     
+    /**
+     * @description Represents a wrapper class used to encapsulate field information.
+     *              This class provides properties to store metadata about a field, such as its name, API name,
+     *              data type, values, sObject name, and relationship name.
+     */
     public class FieldWrap{
+        /**
+         * @description Represents the name of the field.
+         */
         @AuraEnabled
         public String name{get;set;}
+        
+        /**
+         * @description Represents the API name of the field.
+         */
         @AuraEnabled
         public String apiName{get;set;}
+        
+        /**
+         * @description Represents the data type of the field.
+         */
         @AuraEnabled
         public String dataType{get;set;}
+        
+        /**
+         * @description Represents the list of values associated with the field.
+         */
         @AuraEnabled
         public List<String> values{get;set;}
+        
+        /**
+         * @description Represents the API name of the sObject to which the field belongs.
+         */
         @AuraEnabled
         public String sObjectName{get;set;}
+        
+        /**
+         * @description Represents the relationship name of the field (if it's a related field).
+         */
         @AuraEnabled
         public String relationshipName{get;set;}
     }    
+
+     /**
+     * @description    Merges fields from the specified record into the selected template contents
+     * @param          selectedTemplateContents  List of Document_Template_Section__c representing selected template contents
+     * @param          recordIds                  List of Ids of the records from which fields will be merged
+     * @return         List<Document_Template_Section__c> The modified template contents with merged fields
+     */
+    public static List<Document_Template_Section__c> mergefieldsMethodMultiple(List<Document_Template_Section__c> selectedTemplateContents, List<String> recordIds) {   
```

Comment:
The logic used here is similar for the single record, cant we use the mergefieldsMethod by calling it inside a For loop forfetching record specific merge field data?